source/common/http/http3_status_tracker.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "envoy/event/dispatcher.h"
+#include "envoy/event/timer.h"
+
+namespace Envoy {
+namespace Http {
+
+// Tracks the status of HTTP/3 being broken for a period of time
+// subject to exponential backoff.
+class Http3StatusTracker {
+public:
+  Http3StatusTracker(Event::Dispatcher& dispatcher);
+
+  // Returns true if HTTP/3 is broken.
+  bool isHttp3Broken() const;
+  // Returns true if HTTP/3 is confirmed to be working.
+  bool isHttp3Confirmed() const;
```

Comment:
isHttp3Confirmed and markHttp3Confirmed methods seem unused outside of newly added tests.

Reply:
Yes, that's true. Per my previous comment, I'm expecting them to be used in code we don't yet have eventually. We'll want to prevent requests from being sent via 0-RTT to servers that are !isConfirmed(), but we don't have this yet :) We also will want to markConfirmed() when a request succeeds over HTTP/3. But for now we can call this when an HTTP/3 connection succeeds, which is parallel with the code that marks HTTP/3 broken so this seems reasonable for now.